interfaces/prompting/requestprompts/requestprompts.go

```diff
@@ -62,16 +62,20 @@ type PromptConstraints struct {
 	Path                 string   `json:"path"`
 	Permissions          []string `json:"permissions"`
 	AvailablePermissions []string `json:"available-permissions"`
+	// Preserve originally-requested permissions so we can send a response with
+	// explicitly-allowed/denied permissions, even if some permissions were
+	// previously satisfied.
+	originalPermissions []string
```

Comment:
while going back and forth between interfaces/prompting.Constraints and this struct, I noticed that the former does not have a doc comment which would otherwise be useful when one comes back to the code after a while. (otherwise PromptConstraints being line prompting.Constraints isn't very useful if the latter isn't documented).